posts/navarroaxel/10.improve-your-website-performance-using-patytown.md

```diff
@@ -0,0 +1,130 @@
+```
+
+### Where is the `forward`?
+
+To `forward` a function call to the worker we just need to add the configuration to the `window`  object before the inlined Partytown `script`.
+
+```html
+<script>
+  window.partytown = { forward: ['dataLayer.push'] };
+</script>
+<script type="text/partytown">
+  /* GTM script here */
+</script>
+```
+
+Now, you can use `window.dataLayer.push()` from the main thread. 🙌
+
+## Using workers in Next.js
+
+Setting up Partytown in a Next.js website is veryeasy! We just need Next.js [v12.1.1](https://github.com/vercel/next.js/releases/tag/v12.1.1) or later.
+
+🧠 Remember that this is experimental feature in Next.js and it’s not covered by semver. 🙀
+
+First, we should enable the experimental flag in the `next.config.js` file.
+
+```javascript
+module.exports = {
+  experimental: { nextScriptWorkers: true }
+};
+```
+
+And install the Partytown package.
+
+```bash
+npm install --save-dev @builder.io/partytown
+```
+
+The Next.js workers don't require extra configurations, but we should `forward` the `dataLayer.push` for this example. We need to include this configuration in a `<Head>` component inside a custom `_document.js`.
+
+```javascript
+import { Html, Head, Main, NextScript } from 'next/document'
+
+const Document = () => (
+  <Html>
+    <Head>
+      <script
+        data-partytown-config
+        dangerouslySetInnerHTML={{
+          __html: `
+            partytown = {
+              lib: '/_next/static/~partytown/',
+              debug: true,
+              forward: ['dataLayer.push']
+            };
+          `
+        }}
+      />
+    </Head>
+    <body>
+      <Main />
+      <NextScript />
+    </body>
+  </Html>
+);
+
+export default Document;
+```
+
+💡 The `debug: true` setting gives us nice info in the DevTools' console.
+
+Finally, we can add the GTM script in our `_app.js`.
+
+```javascript
+import Script from 'next/script'; 
+
+const MyApp = ({ Component, pageProps }) => (
+  <>
+    <Script
+      id="gtm-script"
+      strategy="worker"
+      dangerouslySetInnerHTML={{
+        __html: `/* GTM script here */`
+      }}
+    />
+    <Component {...pageProps} />
+  </>
+);
+
+export default MyApp;
+```
+
+The `id` property is required for _inline scripts_ in order to let Next.js track and optimize the script.
+
+And it's done! Your GTM runs in a worker that you can communicate with using the `dataLayer.push` - as always.
+
+You can find installation instructions for [React](https://partytown.builder.io/react), [Angular](https://partytown.builder.io/angular), [Nuxt](https://partytown.builder.io/nuxt) and [more](https://partytown.builder.io/integrations).
+
+## Conclusion
+
+Too much JavaScript can block DOM construction, delaying how quickly pages can render. Reducing JavaScript in our main thread is a good strategy to reduce the time until interaction. This will free up main thread resources so they can only be used for the primary web app execution, e.g. network requests.
```

Comment:
```suggestion
Too much JavaScript can block DOM construction, delaying how quickly pages can render. Reducing JavaScript in our main thread is a good strategy to reduce the time until interaction. This will free up main thread resources so they can be used for the primary web app execution, e.g. network requests.
```